test/eqDeep.js

```diff
@@ -0,0 +1,39 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('eqDeep', function() {
+    var a = [];
+    var b = a;
+    var c = {}; c.v = c;
+    var d = {}; d.v = d;
+    var e = []; e.push(e);
+    var f = []; f.push(f);
+    var nestA = {a:[1, 2, {c:1}], b:1};
+    var nestB = {a:[1, 2, {c:1}], b:1};
+    var nestC = {a:[1, 2, {c:2}], b:1};
+    var listA = [1, 2, 3];
+    var listB = [1, 3, 2];
+    it('tests for deep equality of its operands', function() {
+        assert.strictEqual(R.eqDeep(100, 100), true);
+        assert.strictEqual(R.eqDeep(100, '100'), false);
+        assert.strictEqual(R.eqDeep([], []), true);
+        assert.strictEqual(R.eqDeep(a, b), true);
+        assert.strictEqual(R.eqDeep([], {}), false);
+        assert.strictEqual(R.eqDeep(c, d), true);
+        assert.strictEqual(R.eqDeep(e, f), true);
+        assert.strictEqual(R.eqDeep(nestA, nestB), true);
+        assert.strictEqual(R.eqDeep(nestA, nestC), false);
+        assert.strictEqual(R.eqDeep(listA, listB), false);
+    });
+
+    it('is curried', function() {
+        var isA = R.eqDeep(a);
+        assert.strictEqual(isA([]), true);
+    });
+
+    it('throws if given no arguments', function() {
+        assert.throws(R.eqDeep, TypeError);
+    });
+});
```

Comment:
i'd like to see tests that prove that logically equivalent objects are equal without regard to key order, e.g.:

```javascript
var abc = {a: 1, b: 2, c: 3};
var bac = {b: 2, a: 1, c: 3};
var acb = {a: 1, c: 3, b: 2};
// etc.
assert.strictEqual(R.eqDeep(abc, acb));
assert.strictEqual(R.eqDeep(abc, bac));
assert.strictEqual(R.eqDeep(bac, acb));
// etc.
```